beta/src/pages/apis/react/useMemo.md

```diff
@@ -2,17 +2,328 @@
+}
+
+function getVisibleTodos(todos, tab) {
+  console.log("Running getVisibleTodos for " + todos.length + " todos");
+  return todos.filter((t) => {
+    if (tab === "completed") {
+      return t.completed;
+    }
+    return true;
+  });
+}
+
+export default function App() {
+  const [todos, setTodos] = useState(createInitialTodos);
+  const [tab, setTab] = useState("all");
+  const [draft, setDraft] = useState("");
+  const visibleTodos = useMemo(() => getVisibleTodos(todos, tab), [todos, tab]);
+
+  const toggleTodo = useCallback((id) => {
+    setTodos((todos) =>
+      todos.map((t) => {
+        if (t.id === id) {
+          return {
+            ...t,
+            completed: !t.completed
+          };
+        } else {
+          return t;
+        }
+      })
+    );
+  }, []);
+
+  return (
+    <>
+      <button onClick={() => setTab("all")}>All</button>
+      <button onClick={() => setTab("completed")}>Completed</button>
+      <hr />
+      <input
+        value={draft}
+        onChange={(e) => {
+          setDraft(e.target.value);
+        }}
+      />
+      <hr />
+      <TodoList visibleTodos={visibleTodos} toggleTodo={toggleTodo} />
+    </>
+  );
+}
+
+```
+```js TodolistComponent.js
+
+import { memo } from "react";
+
+export default function TodoList({ visibleTodos, toggleTodo }) {
+    console.log("TodoList re-render");
+    return (
+      <ul>
+        {visibleTodos.map((item) => (
+          <li key={item.id}>
+            <label>
+              <input
+                type="checkbox"
+                checked={item.completed}
+                onChange={(e) => {
+                  toggleTodo(item.id);
+                }}
+              />
+              {item.text}
+            </label>
+          </li>
+        ))}
+      </ul>
+    );
+  }
+  TodoList = memo(TodoList);
+
+```
+
+</Sandpack>
+
+<Solution />
+
+</Recipes>
+
+<DeepDive title="Pros and Cons of using `useMemo`">
+Pros
+
+`useMemo` only recomputes the memoized value when one of the dependencies has changed. So, re-computing an expensive function on every render is unnecessary.
+
+Cons
+
+Every time a component is rendered:
+* an extra call is made to `useMemo`, thus creating performance overhead
+* memory gets allocated to hold the memoized variable value, thus creating memory overhead
+
+</DeepDive>
+
+
+## Reference {/*reference*/}
+
+### `useMemo(() => computeExpensiveValue(a, b), [a, b])` {/*usememo*/}
+
+Call `useMemo` at the top level of your component to declare one or more memoized variables.
 
 ```js
-const memoizedValue = useMemo(callback, [...dependencies])
+import {useMemo} from 'react';
+  // ...
 ```
 
-</Intro>
+#### Parameters {/*parameters*/}
+
+* `computeExpensiveValue`: The parameter is the expensive computational function that requires memoization.
+
+* `[a,b]`: This value represents the dependency on which the recompute of the `computeExpensiveValue` rests. There can be an array of dependencies or a single value.
```

Comment:
Should this be `deps`? `[a,b]` is confusing because there's not always two deps.